modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2BackChannelLogoutTestCase.java

```diff
@@ -0,0 +1,269 @@
+        this.activeTenant = context.getContextTenant().getDomain();
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        init();
+        createOIDCApplication();
+        client = HttpClientBuilder.create().setDefaultCookieStore(new BasicCookieStore()).build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testCleanUp() throws Exception {
+
+        removeApplications();
+    }
+
+    protected void init() throws Exception {
+
+        super.init();
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+    }
+
+    @Test(groups = "wso2.is", description = "Test back channel logout for OIDC.")
+    public void testOIDCLogout() throws IOException {
+
+        // Login.
+        initiateOIDCRequest(false);
+        performOIDCLogin();
+        performOIDCConsentApproval();
+
+        // OIDC app logout.
+        performOIDCLogout();
+    }
+
+    private void createOIDCApplication() throws Exception {
+
+        OAuthConsumerAppDTO appDTO = new OAuthConsumerAppDTO();
+        appDTO.setApplicationName(OIDC_APP_NAME);
+        appDTO.setCallbackUrl(OAuth2Constant.CALLBACK_URL);
+        appDTO.setOAuthVersion(OAuth2Constant.OAUTH_VERSION_2);
+        appDTO.setGrantTypes(OAuth2Constant.OAUTH2_GRANT_TYPE_AUTHORIZATION_CODE);
+        appDTO.setBackChannelLogoutUrl("http://localhost:" + DEFAULT_TOMCAT_PORT + "/playground2/bclogout");
+
+        adminClient.registerOAuthApplicationData(appDTO);
+        OAuthConsumerAppDTO createdApp = adminClient.getOAuthAppByName(OIDC_APP_NAME);
+        Assert.assertNotNull(createdApp, "Adding OIDC app failed.");
+        oidcAppClientId = createdApp.getOauthConsumerKey();
+    }
+
+    private void removeApplications() throws Exception {
+
+        adminClient.removeOAuthApplicationData(oidcAppClientId);
+    }
+
+    private void initiateOIDCRequest(boolean isCheckLogoutConfirmation) throws IOException {
+
+        List<NameValuePair> urlParameters = getOIDCInitiationRequestParams();
+        HttpResponse response = sendPostRequestWithParameters(urlParameters,
+                "http://localhost:" + DEFAULT_TOMCAT_PORT + "/playground2/oauth2-authorize-user.jsp");
+        Assert.assertNotNull(response, "Authorization response is null");
+        Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+        Assert.assertNotNull(locationHeader, "Authorization response header is null.");
+        EntityUtils.consume(response.getEntity());
+
+        response = sendGetRequest(locationHeader.getValue());
+        sessionDataKey = Utils.extractDataFromResponse(response, CommonConstants.SESSION_DATA_KEY, 1);
+        if (isCheckLogoutConfirmation) {
+            Assert.assertNotNull(sessionDataKey, "Back channel logout failed for OIDC.");
+        } else {
+            Assert.assertNotNull(sessionDataKey, "sessionDataKey is null for ." + OIDC_APP_NAME);
```

Comment:
Incorrect formatting of the error message

Reply:
Fixed https://github.com/wso2/product-is/pull/9453/commits/836990572286f04de75d9ce9c940e23502842139